app/frontend/src/features/user/UserTabContext.test.tsx

```diff
@@ -0,0 +1,46 @@
+import { useTabContext } from "@material-ui/lab";
+import { render, screen } from "@testing-library/react";
+import React from "react";
+import { MemoryRouter } from "react-router";
+import { Route } from "react-router-dom";
+
+import UserTabContext from "./UserTabContext";
+
+const Test = () => {
+  const tab = useTabContext();
+  return <>{tab?.value}</>;
+};
+
+function renderTest(url: string) {
+  render(
+    <MemoryRouter initialEntries={[url]}>
+      <Route path="profile/:tab?">
+        <UserTabContext>
+          <Test />
+        </UserTabContext>
+      </Route>
+    </MemoryRouter>
+  );
+}
+
+describe("UserTabContext", () => {
+  it("accepts 'home' as tab parameter from url path", () => {
+    renderTest("profile/home");
+    screen.getByText("home");
```

Comment:
```suggestion
    expect(screen.getByText("home")).toBeInTheDocument();
```
Same comment for the others so we get better error messages for when the tests fail in future for whatever reason (e.g. we changed something or broken something unintentionally)

Reply:
Is that needed? The jest docs write that getByText throws an error with a descriptive error message:
https://testing-library.com/docs/queries/about/#types-of-queries

> getBy...: Returns the matching node for a query, and throw a descriptive error if no elements match or if more than one match is found (use getAllBy instead if more than one element is expected).

Reply:
screen.getByText("foobar")
```javascript
 FAIL  src/features/user/UserTabContext.test.tsx
  UserTabContext
    ✓ accepts 'home' as tab parameter from url path (40 ms)
    ✓ accepts 'about' as tab parameter from url path (5 ms)
    ✓ replaces invalid tab parameters from url path with default 'about' (5 ms)
    ✕ uses the default 'about' when there is no tab parameter (9 ms)

  ● UserTabContext › uses the default 'about' when there is no tab parameter

    TestingLibraryElementError: Unable to find an element with the text: foobar. This could be because the text is broken up by multiple elements. In this case, you can provide a function for your text matcher to make your matcher more flexible.

    <body>
      <div>
        about
      </div>
    </body>

      42 |   it("uses the default 'about' when there is no tab parameter", () => {
      43 |     renderTest("profile");
    > 44 |     screen.getByText("foobar");
         |            ^
      45 |   });
      46 | });
      47 |

      at Object.getElementError (node_modules/@testing-library/dom/dist/config.js:37:19)
      at node_modules/@testing-library/dom/dist/query-helpers.js:90:38
      at node_modules/@testing-library/dom/dist/query-helpers.js:62:17
      at getByText (node_modules/@testing-library/dom/dist/query-helpers.js:111:19)
      at Object.<anonymous> (src/features/user/UserTabContext.test.tsx:44:12)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 3 passed, 4 total
Snapshots:   0 total
Time:        4.077 s
```

expect(screen.getByText("foobar").toBeInTheDocument())
```javascript
 FAIL  src/features/user/UserTabContext.test.tsx
  UserTabContext
    ✓ accepts 'home' as tab parameter from url path (33 ms)
    ✓ accepts 'about' as tab parameter from url path (5 ms)
    ✓ replaces invalid tab parameters from url path with default 'about' (6 ms)
    ✕ uses the default 'about' when there is no tab parameter (8 ms)

  ● UserTabContext › uses the default 'about' when there is no tab parameter

    TestingLibraryElementError: Unable to find an element with the text: foobar. This could be because the text is broken up by multiple elements. In this case, you can provide a function for your text matcher to make your matcher more flexible.

    <body>
      <div>
        about
      </div>
    </body>

      42 |   it("uses the default 'about' when there is no tab parameter", () => {
      43 |     renderTest("profile");
    > 44 |     expect(screen.getByText("foobar")).toBeInTheDocument();
         |                   ^
      45 |   });
      46 | });
      47 |

      at Object.getElementError (node_modules/@testing-library/dom/dist/config.js:37:19)
      at node_modules/@testing-library/dom/dist/query-helpers.js:90:38
      at node_modules/@testing-library/dom/dist/query-helpers.js:62:17
      at getByText (node_modules/@testing-library/dom/dist/query-helpers.js:111:19)
      at Object.<anonymous> (src/features/user/UserTabContext.test.tsx:44:19)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 3 passed, 4 total
Snapshots:   0 total
Time:        3.87 s
```

Reply:
Ahh interesting, it looks like the error comes from the same place anyway so that's good to know!

In that case, I think it's more of a style nitpick for me, since imo it's unclear on its own that we are meant to be asserting something with that statement as I could also write that on its own to wait for something if I don't care about the result. Putting things in `expect(...).<assertion>` makes it clearer it's a test result we care about.

Reply:
I thought I had once read that wrapping it with expect was bad practice. However, [this article](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library) - although the author agrees it's opiniated - claims the opposite. And since the other remarks of this article are already taken care of very well here in the couchers repo, I'll hapilly add the assertions back.

(Except for the one regarding the discussions test because that one really always fails the test for me :( )

Reply:
I thought I had once read that wrapping it with expect was bad practice. However, [this article](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library) - although the author agrees it's opiniated - claims the opposite. And since the other remarks of this article are already taken care of very well here in the couchers repo, I'll hapilly add the assertions back.

(Except for the one regarding the discussions test because that one really always fails the test for me :( )

Reply:
It's a good article since React Testing Library is written/started by Kent! The tests in this repo have intentionally followed the medium to high importance tips he's given there as after all, who knows better how to use the library than the library author himself :)

This diff here still says you haven't update these though? 🤔